File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l,r=0,0
        n=len(numbers)
        ans=[]
        while l<n:
            r=l
            while r<n:
                if numbers[l]+numbers[r]==target:
                    ans.append(l+1)
                    ans.append(r+1)
                    return ans
                elif numbers[l]+numbers[r]>target:
                    break
                else:
                    r+=1
            l+=1
        return ans

File: Data Structures & Algorithms/search-2d-matrix/submission-1.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int m=matrix.size();
        int n=matrix[0].size();

        int i=0;

        while (i<m){
            if (matrix[i][0]<=target && target<=matrix[i][n-1]){
                int l=0,r=n-1;
                int mid=l+(r-l)/2;
                while(l<=r){
                    if(matrix[i][mid]==target){
                        return true;
                    }
                    else if(matrix[i][mid]<target){
                        l=mid+1;
                    }
                    else{
                        r=mid-1;
                    }
                    mid = l + (r - l) / 2;
                }
            }
            i+=1;

        } 
        return false;     
    }
};
